esp/esp/program/modules/handlers/lotteryfrontendmodule.py

```diff
@@ -0,0 +1,98 @@
+from esp.program.models import Program, ClassSection
+from esp.program.modules.base import ProgramModuleObj, needs_admin, main_call, aux_call
+from esp.program.controllers.lottery import LotteryAssignmentController, LotteryException
+from esp.utils.web import render_to_response
+from esp.users.models import ESPUser
+from esp.utils.decorators import json_response
+import numpy
+
+class LotteryFrontendModule(ProgramModuleObj):
+
+    @classmethod
+    def module_properties(cls):
+        return {
+            "admin_title": "Lottery Frontend",
+            "link_title": "Run the Lottery Assignment Thing",
+            "module_type": "manage",
+            "seq": 10
+            }
+
+    @main_call
+    @needs_admin
+    def lottery(self, request, tl, one, two, module, extra, prog):
+        #   Check that the lottery module is included
+        students = self.program.students()
+        if 'lotteried_students' not in students and 'twophase_star_students' not in students:
+            return render_to_response(self.baseDir() + 'not_configured.html', request, {'program': prog})
+
+        #   Render control page with lottery options
+        context = {'options': {k: v for k, v in LotteryAssignmentController.default_options.items() if v[1] is not False}}
+        return render_to_response(self.baseDir()+'lottery.html', request, context)
+
+    def is_float(self, s):
+        try:
+            float(s)
+            return True
+        except ValueError:
+            return False
+
+    @aux_call
+    @json_response()
+    @needs_admin
+    def lottery_execute(self, request, tl, one, two, module, extra, prog):
+        # find what options the user wants
+        options = {}
+
+        for key in request.POST:
+            if 'lottery_' in key:
+                value = request.POST[key]
+
+                if value == 'True':
+                    value = True
+                elif value == 'False':
+                    value = False
+                elif value == 'None':
+                    value = None
+                elif self.is_float(value):
+                    value = float(value)
+
+                options[key.split('_', 1)[1]] = value
+
+        error_msg = ''
+        try:
```

Comment:
Is there a reason to have the two identical `try`/`except` blocks instead of just one?